database/database_handler.py

```diff
@@ -111,3 +93,48 @@ def managed_session(self):
             raise e
         finally:
             session.close()
+
+    def close_connection(self):
+        try:
+            # Dispose the engine
+            self.engine.dispose()
+        except SQLAlchemyError as e:
+            LOGGER.info(f"Error disposing the database engine: {str(e)}")
+            raise e
+
+    @staticmethod
+    def extract_upload_date(path):
+        parts = path.split('/')
+        image_filename = parts[-1]
+        date_time_str = parts[-2]
+
+        try:
+            image_upload_date = datetime.strptime(date_time_str, "%Y-%m-%d_%H:%M:%S")
+            image_upload_date = image_upload_date.strftime("%Y-%m-%d")
+        except ValueError as e:
+            LOGGER.info(f"Invalid folder structure, can not retrieve date: {date_time_str}")
+            raise e
+
+        return image_filename, image_upload_date
+
+    def _renew_token_periodically(self):
+        while True:
```

Comment:
A bit scared of this way I have to say. 
Cannot we just do the check when we use the database handler?
For example when querying we check if the token is expired and then we query.

Reply:
We do a lot of querying, checken each time the things we do in _renew_token_periodically will give a lot of overhead. I will think about other options